python-avd/pyavd/_eos_designs/shared_utils/mlag.py

```diff
@@ -77,15 +87,25 @@ def mlag_peer_l3_ipv6_pool(self: SharedUtilsProtocol) -> str:
     @cached_property
     def mlag_role(self: SharedUtilsProtocol) -> Literal["primary", "secondary"] | None:
         # Note: self.node_group_is_primary_and_peer_hostname is always set when self.mlag is true, so this is just to make type-checker happy.
-        if self.mlag and self.node_group_is_primary_and_peer_hostname is not None:
+        if not self.mlag:
+            return None
+        if self.node_group_is_primary_and_peer_hostname is not None:
             return "primary" if self.node_group_is_primary_and_peer_hostname[0] else "secondary"
 
+        if self.switch_facts.mlag_peer:
+            return "primary" if natural_sort([self.hostname, self.switch_facts.mlag_peer])[0] == self.hostname else "secondary"
```

Comment:
would this not work for both cases?

Reply:
the historic behavior is to take the first device in the node_group. Now we sort them and take the first from the sorted list. This will not give a difference in most cases, but someone might be using asset numbers as hostnames, and then this could give a diff.